### ISSUE TITLE
training: SFT training

### ISSUE THREAD

**Issue by tikikun** [menloresearch/space-thinker#5]

**Comment by bachvudinh** [comment]
# Goal
- To train a model, inspired by AlphaMaze, to master manipulation tasks within a table-top environment, specifically targeting the EmbodiedBench benchmark.

# Methodology
- A two-stage synthetic dataset generation process was employed:

  a. Object Placement:
  Objects were randomly positioned on the table surface. Coordinates (x, y) were uniformly sampled from the range [10, 90] (units unspecified, but presumably consistent with the EmbodiedBench environment). A minimum inter-object distance constraint was enforced to prevent unrealistic overlaps. Object heights (z) were uniformly sampled from the range [17, 19].
  
  b. Reasoning Data Generation:
  A structured reasoning dataset was created to guide the model's training. The format for thinking part:
 ```
  LOCATE OBJECTS:
  Target Object: {source_object} found at {source_pos} with height {source_height}
  Target Placement Location: {target_object} at {target_pos} with height {target_height}
  PLAN ACTIONS:
  Step 1: Move above {source_object} at {source_pos} with height 30
  Step 2: Lower to {source_object} at {source_pos} with height {source_height}
  Step 3: Close gripper on {source_object} at {source_pos} with height {source_height}
  Step 4: Lift {source_object} at {source_pos} to height 30
  Step 5: Move to {target_object} location at {target_pos} with height 30
  Step 6: Stack on top of {target_object} location at {target_pos} with height {target_height} + 4 and open gripper
 ```

## Experiments 
- Visual Thinker model

| Run ID | Date       | Model Config | Dataset | Learning Rate | Batch Size | Steps | Loss | Hardware | 
|--------|------------|--------------|---------|---------------|------------|-------|------|----------|
| exp-1 | 2025-03-13 | [Full-Finetune](https://github.com/janhq/pick-place-model/blob/main/training_config.yaml) | [Pick-Place-Table](https://huggingface.co/datasets/jan-hq/Pick-Place-Table-Reasoning) | 1e-4 | 64  |  |   0.045| ~ 1.5 hours on 6xA6000s | 
| exp-2 | 2025-03-14 | [Full-Finetune](https://github.com/janhq/pick-place-model/blob/main/training_config.yaml) | [Pick-Place-Table-Reasoning](https://huggingface.co/datasets/jan-hq/Maze-Reasoning) | 1e-4 |  64 |  |   0.03 | ~ 6 hours on 6xA6000s |

**Comment by tikikun** [comment]
![Image](https://github.com/user-attachments/assets/a9493a36-2804-419e-a021-cd4643f6dba1)

**Comment by bachvudinh** [comment]
New training run on 8xH200:
- Goal: Model has ability to map the action output from 25x25 space into 100x100 space directly.
- Methodology: Change the data a little bit that the final output is a 7D discrete gripper action in the following format: [X_100, Y_100, Z, Roll, Pitch, Yaw, Gripper] with X and Y are discrete positions on a 100×100 grid representing the table surface.
- Time taken: 40 minutes.
- Loss:
  ![Image](https://github.com/user-attachments/assets/3a6e8093-929a-4167-b7a8-970b69705d53)

- Key Issues Identified
  + Hallucination: Model produces coordinates that don't correspond to actual objects
  + Object localization: Model struggles to accurately identify object positions
  + No demonstrated capability to successfully map from 25×25 to 100×100 coordinate space

**Comment by bachvudinh** [comment]
## New data strat:
 - Adding local position information to help reconstruct from 25x25 to 100x100 space.
 - Spawn multiple containers for Placing task instead of one for now.
 - Fixing the z coordinate.
 - Make some change for thinking part of data.
 - Introducing moving data: that required model to move a  object to a specific given [x,y,z] coordinate.
## Experiments 
- Space Thinker model

| Run ID | Date       | Model Config | Dataset | Learning Rate | Batch Size | Steps | Loss | Hardware | 
|--------|------------|--------------|---------|---------------|------------|-------|------|----------|
| exp-adding-local-info | 2025-03-18 | [Full-Finetune](https://github.com/janhq/pick-place-model/blob/main/training_config.yaml) | [Pick-Place-Table-with-Local-Pos](https://huggingface.co/datasets/jan-hq/Pick-Place-Table-Reasoning-local-pos) | 1e-4 | 128  |  |   0.045| ~ 40 mins on H200  | 
| exp-adding-local-info | 2025-03-18 | [Full-Finetune](https://github.com/janhq/pick-place-model/blob/main/training_config.yaml) | [Pick-Place-Table-with-Local-Pos-v0.2](https://huggingface.co/datasets/jan-hq/Pick-Place-Table-Reasoning-local-pos-v0.2) | 1e-4 | 128  |  |   0.045| ~ 40 mins on H200  | 

## Evaluate Resust on EmodiedBench- Manipulation Subtask

| Model | Picking | Stacking | Total (%) |
|-------|---------|----------|-----------|
| AlphaSpace (Ours) | [10/12](https://github.com/user-attachments/files/19399135/planner_output.json) | [6/12](https://github.com/user-attachments/files/19399169/planner_output_stacking.json) | 66.67% |
| GPT-4o | 6/12 | 3/12 | 37.5% |
| Claude 3.5 Sonnet | 5/12 | 2/12 | 29.17% |
